### ISSUE TITLE
Make K Means an Online Learner with inertia

### ISSUE THREAD

**Issue by andrewdalpino** [RubixML/ML#20]
K Means is a well-known clusterer used throughout industry due to its speed and accuracy. The current implementation of K Means uses mini batch gradient descent with the stopping criteria being if the number of sample training points that change cluster mappings over an epoch is below a threshold then stop. Mini batch gradient descent is often used in Online learners as a way to iteratively train a model over time. However, the current stopping criteria becomes non-intuitive in the context of partial training. Specifically, we'd like the stop to be based on some minimum change in the loss function.

Thus, I propose to use the inertial cost function a.k.a. the sum of distances between each training sample and its nearest cluster centroid as a basis for a new stopping criteria which would enable online learning.
